packages/experiment-codec/tests/testVector.test.ts

```diff
@@ -0,0 +1,266 @@
+      f2: schema.StructA,
+      f3: schema.Bytes,
+      f4: schema.BytesVec,
+      f5: schema.Table1,
+      f6: schema.BytesOpt,
+      f7: schema.UnionA,
+      f8: schema.byte,
+    };
+
+    if (testCase.data) {
+      //  @ts-ignore
+      Object.entries(testCase.data).forEach((testDataEntry) => {
+        const [itemCodecName, itemValue] = testDataEntry;
+        let itemCodec: BytesCodec;
+        if (testName === "Table5") {
+          // @ts-ignore
+          itemCodec = itemCodecTable5Map[itemCodecName];
+        } else if (testName === "TableA") {
+          // @ts-ignore
+          itemCodec = itemCodecTableAMap[itemCodecName];
+        } else {
+          throw new Error("Not implemented test case.");
+        }
+        // @ts-ignore
+        testData[itemCodecName] = itemCodec.unpack(toArrayBuffer(itemValue));
+      });
+    }
+  } else if (testName.match(/Opt$/)) {
+    // option ByteOpt (byte);
+    // option WordOpt (Word);
+    // option StructAOpt (StructA);
+    // option StructPOpt (StructP);
+    // option BytesOpt (Bytes);
+    // option WordsOpt (Words);
+    // option BytesVecOpt (BytesVec);
+    // option WordsVecOpt (WordsVec);
+    // option Table0Opt (Table0);
+    // option Table6Opt (Table6);
+    // option Table6OptOpt (Table6Opt);
+    const itemCodecMap = {
+      ByteOpt: schema.byte,
+      WordOpt: schema.Word,
+      StructAOpt: schema.StructA,
+      StructPOpt: schema.StructP,
+      BytesOpt: schema.Bytes,
+      WordsOpt: schema.Words,
+      BytesVecOpt: schema.BytesVec,
+      WordsVecOpt: schema.WordsVec,
+      Table0Opt: schema.Table0,
+      Table6Opt: schema.Table6,
+      Table6OptOpt: schema.Table6Opt,
+    };
+    //  @ts-ignore
+    const itemCodec = itemCodecMap[testName];
+    //  @ts-ignore
+    testData = testCase.item;
+    if (testCase.item) {
+      testData = itemCodec.unpack(toArrayBuffer(testCase.item));
+    }
+  } else if (testName === "UnionA") {
+    // union UnionA {
+    //     byte,
+    //     Word,
+    //     StructA,
+    //     Bytes,
+    //     Words,
+    //     Table0,
+    //     Table6,
+    //     Table6Opt,
+    // }
+    const itemCodecMap = {
+      byte: schema.byte,
+      Word: schema.Word,
+      StructA: schema.StructA,
+      Bytes: schema.Bytes,
+      Words: schema.Words,
+      Table0: schema.Table0,
+      Table6: schema.Table6,
+      Table6Opt: schema.Table6Opt,
+    };
+    //  @ts-ignore
+    const itemCodec = itemCodecMap[testCase.item.type];
+    //  @ts-ignore
+    testData = itemCodec.unpack(toArrayBuffer(testCase.item.data));
+    testData = { type: testCase.item.type, value: testData };
+  } else {
+    console.warn(`WARNING: ${testName} is not tested`);
+  }
+  const packed = codec.pack(testData);
+  test.serial(
+    `TestName: ${testName}, No.${tag}:\n 
+    Test case is: ${JSON.stringify(testCase)}\n 
+    Unpacked: ${JSON.stringify(testData)}\n`,
+    (t) => {
+      t.deepEqual(toHex(packed), testCase.expected);
+    }
```

Comment:
`test.serial(...)` breaks ci output, we can just `test("should same with expected when packing data in simple.yaml")`

Reply:
The title for each test case should differ from each other. Added index to achieve this.